src/generators/lua_map_generator.cpp

```diff
@@ -37,15 +37,16 @@ lua_map_generator::lua_map_generator(const config & cfg, const config* vars)
 	, generator_data_(cfg)
 {
 	lk_.load_core();
-	const std::array required {"id", "config_name", "create_map"};
-	for(const std::string& req : required) {
+	using namespace std::string_literals;
+	const std::array required {"id"s, "config_name"s, "create_map"s};
+	for(const auto& req : required) {
 		if (!cfg.has_attribute(req)) {
 			if(req == "create_map" && cfg.has_attribute("create_scenario")) {
 				// One of these is required, but not both
 				continue;
 			}
 			std::string msg = "Error when constructing a lua map generator -- missing a required attribute '";
-			msg += req + "'\n";
+			msg += req + std::string("'\n");
```

Comment:
Don't need the extra string construction but I guess it can't hurt.

Reply:
Oh, I suppose that's true now that `req` isn't a `const char*`.